Changed Notify Function

MarkSkith opened the pull request:
Added a timer parameter to the function notify to allow for autoclosing notification prompts, I also added an if statement to check if callbacks were created in order to determine whether or not to show the accept and decline buttons.

MarkSkith committed:
Changed Notify Function

Added a timer parameter to the function notify to allow for autoclosing notification prompts, I also added an if statement to check if callbacks were created in order to determine whether or not to show the accept and decline buttons.

GreenDeno commented on source.lua:
```diff
@@ -499,7 +499,7 @@ do
 	-- new modules
 	
-	function library:Notify(title, text, callback)
+	function library:Notify(title, text, timer, callback)
```
The main issue I see here is that you are changing a parameter, not backward compatible. Add the timer parameter after callback.

GreenDeno commented on source.lua:
```diff
@@ -657,6 +657,16 @@ do
+			notification.Accept.Visible = false
+			end
+		
+		if timer > 0 then
```
Timer should be optional, so you should check if it exists.

GreenDeno commented on source.lua:
```diff
@@ -657,6 +657,16 @@ do
+			end
+		
+		if timer > 0 then
+			wait(timer)
```
The yield should be on another thread as it should not halt the current thread's execution.

MarkSkith committed:
Update source.lua

MarkSkith commented:
Fixed

MarkSkith committed:
Changed notification decline/accept

If callback doesn't exist but timer does exist(to prevent showing a notification that cannot be closed.) then show a accept only, as there isn't a difference between decline and accept if the dev did not create a callback(lowers confusion about the difference between accept and decline for users of this library)
